### PR TITLE
Update Prometheus to v2.19.2

### LINKED ISSUE #1124: Update Prometheus to v2.19.0
**Is your feature request related to a problem? Please describe.**
OpenTelemetry Collector currently depends on Prometheus v2.13.0 (September 2019), which is missing some of the features and bugfixes from the latest v2.19.0 release.

**Describe the solution you'd like**
- Update `go.mod` dependency to v2.19.0 (it's written as`v1.8.2-0.20200609102542-5d7e3e970602` by `go mod`)
- Fix breaking changes in Prometheus Receiver (`Appender` interface)
- Fix conflicting dependencies: see https://github.com/uber/tchannel-go/issues/781 - **this a blocker, breaking Jaeger receiver**

**Additional context**
One specific feature we need is support for `selectors:` section in the Kubernetes discovery config for Prometheus, which was introduced in Prometheus v2.17.0: https://github.com/prometheus/prometheus/pull/6807

### PR DESCRIPTION
**Pull request by nilebox**
**Description:** 
Update prometheus to v2.19.2

**Link to tracking Issue:** Fixes #1124

**Testing:** Existing tests should cover potential regression

**Documentation:** -

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -108,3 +108,8 @@ issues:
     - path: _test\.go
       linters:
         - scopelint
+    # Exclude some staticcheck messages
+    - linters:
+      - staticcheck
+      # See https://github.com/golang/protobuf/issues/1077
```
Comment (nilebox) [inline .golangci.yml:114]: Updated Prometheus also brought `protobuf v1.4.1`, which has `// Deprecated` comment in
https://github.com/golang/protobuf/blob/d04d7b157bb510b1e0c10132224b616ac0e26b17/proto/proto.go#L12
which was introduced in https://go-review.googlesource.com/c/protobuf/+/213901

The change below completely disables the following check:
```
SA1019 | Using a deprecated function, variable, constant or field
```
Is there any better workaround?

File: receiver/prometheusreceiver/internal/ocastore.go [hunk 2]
```diff
@@ -41,7 +40,7 @@ var noop = &noopAppender{}
 
 // OcaStore is an interface combines io.Closer and prometheus' scrape.Appendable
 type OcaStore interface {
-	scrape.Appendable
+	storage.Appendable
```
Comment (nilebox) [inline receiver/prometheusreceiver/internal/ocastore.go:43]: This interface changed required manual implementation changes below.

File: receiver/prometheusreceiver/internal/transaction.go [hunk 3]
```diff
@@ -88,35 +89,33 @@ func newTransaction(ctx context.Context, jobsMap *JobsMap, useStartTimeMetric bo
...
-// there's no document on the first return value, however, it's somehow used in AddFast. I assume this is like a
-// uniqKey kind of thing for storage like a database, so that the operation can be perform faster with this key.
-// however, in this case, return 0 like what the prometheus remote store does shall be enough
-func (tr *transaction) Add(l labels.Labels, t int64, v float64) (uint64, error) {
-	return 0, tr.AddFast(l, 0, t, v)
-}
-
-// returning an error from this method can cause the whole appending transaction to be aborted and fail
-func (tr *transaction) AddFast(ls labels.Labels, _ uint64, t int64, v float64) error {
+// always returns 0 to disable label caching
```
Comment (nilebox) [inline receiver/prometheusreceiver/internal/transaction.go:92]: See https://github.com/prometheus/prometheus/blob/v2.19.2/storage/interface.go#L134
Returning 0 should lead to `AddFast` never being used.

File: .golangci.yml (hunk 1 above)
Comment (nilebox) [reply .golangci.yml:114]: Extracted to #1308 since it's not related to Prometheus itself.

**Comment by jrcamp** [comment]
@nilebox is this still WIP or ready for review?

**Comment by nilebox** [comment]
@jrcamp I will finish this PR once #1308 is merged (it's a blocker).

**Comment by bogdandrutu** [comment]
@nilebox can you work on this?

**Comment by nilebox** [comment]
@bogdandrutu yep, I will update this PR and see if there are any other problems blocking it.

**Pull request merged**